idx/memory/memory.go

```diff
@@ -399,7 +399,9 @@ func (m *MemoryIdx) Load(defs []idx.MetricDefinition) int {
 		m.add(def)
 
 		if TagSupport {
-			m.indexTags(def)
+			// create new nDef to avoid holding open the backing array of defs which is passed up from cassandra / bigtable
```

Comment:
don't refer to concrete interface implementations. just say "persistent index" or something